eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/MgmtLongRunningOperationProvider.cs

```diff
@@ -0,0 +1,226 @@
+                    this, XmlDocProvider.InheritDocs);
+            var waitForCompletionResponseWithPollingAsyncMethod = _isGeneric
+                ? new MethodProvider(
+                    new MethodSignature("WaitForCompletionAsync", null, MethodSignatureModifiers.Public | MethodSignatureModifiers.Override, new CSharpType(typeof(ValueTask), new CSharpType(typeof(Response), _t)), null, [timeSpanParameter, cancellationTokenParameter]),
+                    _operationField.Invoke("WaitForCompletionAsync", [timeSpanParameter, cancellationTokenParameter]),
+                    this, XmlDocProvider.InheritDocs)
+                : new MethodProvider(
+                    new MethodSignature("WaitForCompletionResponseAsync", null, MethodSignatureModifiers.Public | MethodSignatureModifiers.Override, new CSharpType(typeof(ValueTask), typeof(Response)), null, [timeSpanParameter, cancellationTokenParameter]),
+                    _operationField.Invoke("WaitForCompletionResponseAsync", [timeSpanParameter, cancellationTokenParameter]),
+                    this, XmlDocProvider.InheritDocs);
+            return [BuildGetOperationIdMethod(), getRehydrationTokenMethod, getRawResponseMethod, updateStatusMethod, updateStatusAsyncMethod, waitForCompletionResponseMethod, waitForCompletionResponseWithPolingMethod, waitForCompletionResponseAsyncMethod, waitForCompletionResponseWithPollingAsyncMethod];
+        }
+
+        private const string GetOperationIdName = "GetOperationId";
+        private MethodProvider BuildGetOperationIdMethod()
+        {
+            var rehydrationTokenParaemter = new ParameterProvider("rehydrationToken", $"", new CSharpType(typeof(RehydrationToken), true));
+            var signature = new MethodSignature(GetOperationIdName, null, MethodSignatureModifiers.Private, typeof(string), null, [rehydrationTokenParaemter]);
+            var body = new MethodBodyStatement[]
+            {
+                new IfStatement(rehydrationTokenParaemter.Is(Null))
+                {
+                    Return(Null)
+                },
+                Declare("lroDetails", typeof(Dictionary<string, string>), Static(typeof(ModelReaderWriter)).Invoke("Write", [rehydrationTokenParaemter, Static(typeof(ModelReaderWriterOptions)).Property("Json")]).Invoke("ToObjectFromJson", [], new List<CSharpType>{ typeof(Dictionary<string, string>) }, false), out var lroDetailsVariable),
+                Return(new IndexerExpression(lroDetailsVariable, Literal("id")))
+            };
+            return new MethodProvider(signature, body, this);
+        }
+
+        protected override ConstructorProvider[] BuildConstructors()
+            => [ConstructorProviderHelper.BuildMockingConstructor(this), BuildRehydrationConstructor(), BuildInitializationConstructor()];
+
+        private ConstructorProvider BuildInitializationConstructor()
+        {
+            var sourceParameter = new ParameterProvider("source", $"", new CSharpType(typeof(IOperationSource<>), _t));
```

Comment:
Can we add descriptions?

Reply:
added, but not in use for internal methods.

Reply:
We should fix that in MGC. For properties we do generate internal docs.

Reply:
https://github.com/microsoft/typespec/issues/5895